runtime/src/util/set_multimap.rs

```diff
@@ -1,126 +1,162 @@
+    ) -> Result<Self, ActorError> {
+        Ok(Self {
+            outer: Map2::load(bs, root, config.outer, name)?,
+            inner_config: config.inner,
+            value_type: Default::default(),
+        })
     }
 
     /// Retrieve root from the SetMultimap.
     #[inline]
-    pub fn root(&mut self) -> Result<Cid, Error> {
-        self.0.flush()
+    pub fn flush(&mut self) -> Result<Cid, ActorError> {
+        self.outer.flush()
     }
 
     /// Puts the DealID in the hash set of the key.
-    pub fn put(&mut self, key: ChainEpoch, value: DealID) -> Result<(), Error> {
+    pub fn put(&mut self, key: &K, value: V) -> Result<(), ActorError> {
         // Get construct amt from retrieved cid or create new
-        let mut set = self.get(key)?.unwrap_or_else(|| Set::new(self.0.store()));
+        let mut set = self.get(key)?.unwrap_or_else(|| {
+            Set::empty(self.outer.store().clone(), self.inner_config.clone(), "multimap inner")
+        });
 
-        set.put(u64_key(value))?;
+        set.put(&value)?;
 
         // Save and calculate new root
-        let new_root = set.root()?;
+        let new_root = set.flush()?;
 
         // Set hamt node to set new root
-        self.0.set(u64_key(key as u64), new_root)?;
+        self.outer.set(key, new_root)?;
         Ok(())
     }
 
     /// Puts slice of DealIDs in the hash set of the key.
-    pub fn put_many(&mut self, key: ChainEpoch, values: &[DealID]) -> Result<(), Error> {
+    pub fn put_many(&mut self, key: &K, values: &[V]) -> Result<(), ActorError> {
         // Get construct amt from retrieved cid or create new
-        let mut set = self.get(key)?.unwrap_or_else(|| Set::new(self.0.store()));
+        let mut set = self.get(key)?.unwrap_or_else(|| {
+            Set::empty(self.outer.store().clone(), self.inner_config.clone(), "multimap inner")
+        });
 
-        for &v in values {
-            set.put(u64_key(v))?;
+        for v in values {
+            set.put(v)?;
         }
 
         // Save and calculate new root
-        let new_root = set.root()?;
+        let new_root = set.flush()?;
 
         // Set hamt node to set new root
-        self.0.set(u64_key(key as u64), new_root)?;
+        self.outer.set(key, new_root)?;
         Ok(())
     }
 
     /// Gets the set at the given index of the `SetMultimap`
     #[inline]
-    pub fn get(&self, key: ChainEpoch) -> Result<Option<Set<'a, BS>>, Error> {
-        match self.0.get(&u64_key(key as u64))? {
-            Some(cid) => Ok(Some(Set::from_root(*self.0.store(), cid)?)),
+    pub fn get(&self, key: &K) -> Result<Option<Set<BS, V>>, ActorError> {
+        match self.outer.get(key)? {
+            Some(cid) => Ok(Some(Set::load(
+                self.outer.store().clone(),
+                cid,
+                self.inner_config.clone(),
+                "multimap inner",
+            )?)),
             None => Ok(None),
         }
     }
 
     /// Removes a DealID from a key hash set.
```

Comment:
```suggestion
    /// Removes a value from a key hash set.
```

also can we rename `v` to `value` in here for consistency and to make the document match (so we don't have to say "Removes a v from a key hash set"